securebanking-openbanking-uk-rcs-api/src/main/java/com/forgerock/securebanking/openbanking/uk/rcs/factory/AccountConsentDetailsFactory.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.securebanking.openbanking.uk.rcs.factory;
+
+import com.forgerock.securebanking.common.openbanking.uk.forgerock.datamodel.account.FRExternalPermissionsCode;
+import com.forgerock.securebanking.openbanking.uk.rcs.api.dto.consent.details.AccountsConsentDetails;
+import com.forgerock.securebanking.platform.client.IntentType;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import org.joda.time.Instant;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.forgerock.securebanking.openbanking.uk.rcs.api.dto.consent.details.ConsentDetailsConstants.Intent.Members.*;
+import static com.forgerock.securebanking.openbanking.uk.rcs.api.dto.consent.details.ConsentDetailsConstants.Intent.OB_INTENT_OBJECT;
+import static com.forgerock.securebanking.openbanking.uk.rcs.json.utils.JsonUtilValidation.isNotNull;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Account consent details factory implements {@link ConsentDetailsFactory}
+ */
+@Component
+public class AccountConsentDetailsFactory implements ConsentDetailsFactory<AccountsConsentDetails> {
+
+    private final AccountsConsentDetails details;
+
+    @Autowired
+    public AccountConsentDetailsFactory(AccountsConsentDetails details) {
+        this.details = details;
+    }
```

Comment:
The details field is not needed.

The decode method must create a new AccountsConsentDetails object per call for thread safety. The factory produces the ConsentDetails, spring doesn't need to know about them.

Reply:
Deleted the field and create the instance of *ConsentDetails in decode method for thread safety.